File: P622 Design Circular Queue/Solution.java
class MyCircularQueue {
    private Queue<Integer> queue;
    private int limit;
    private int lastAddedElement;

    public MyCircularQueue(int k) {
        queue = new LinkedList<>();
        limit = k;
    }
    
    public boolean enQueue(int value) {
        if (queue.size() < limit) {
            queue.add(value);
            lastAddedElement = value;
            return true;
        }
        return false;
    }
    
    public boolean deQueue() {
        if (queue.size() > 0) {
            queue.poll();
            return true;
        }
        return false;
    }
    
    public int Front() {
        if(queue.size() == 0) {
            return -1;
        }
        return queue.peek();
    }
    
    public int Rear() {
        if(queue.size() == 0) {
            return -1;
        }
        return lastAddedElement;
    }
    
    public boolean isEmpty() {
        return queue.size() == 0;
    }
    
    public boolean isFull() {
        return queue.size() == limit;
    }
}

/**
 * Your MyCircularQueue object will be instantiated and called as such:
 * MyCircularQueue obj = new MyCircularQueue(k);
 * boolean param_1 = obj.enQueue(value);
 * boolean param_2 = obj.deQueue();
 * int param_3 = obj.Front();
 * int param_4 = obj.Rear();
 * boolean param_5 = obj.isEmpty();
 * boolean param_6 = obj.isFull();
 */

File: P622 Design Circular Queue/solution.py
class MyCircularQueue:

    def __init__(self, k: int):
        self.queue = []
        self.limit = k
        

    def enQueue(self, value: int) -> bool:
        if len(self.queue) < self.limit:
            self.queue.append(value)
            return True
        return False
        

    def deQueue(self) -> bool:
        if len(self.queue) > 0:
            self.queue.pop(0)
            return True
        return False
        

    def Front(self) -> int:
        if len(self.queue) != 0:
            return self.queue[0]
        return -1
        

    def Rear(self) -> int:
        if len(self.queue) != 0:
            return self.queue[-1]
        return -1

    def isEmpty(self) -> bool:
        return len(self.queue) == 0
        

    def isFull(self) -> bool:
        return len(self.queue) == self.limit


# Your MyCircularQueue object will be instantiated and called as such:
# obj = MyCircularQueue(k)
# param_1 = obj.enQueue(value)
# param_2 = obj.deQueue()
# param_3 = obj.Front()
# param_4 = obj.Rear()
# param_5 = obj.isEmpty()
# param_6 = obj.isFull()